config/routes.rb

```diff
@@ -2,7 +2,11 @@
   scope '(:locale)', locale: /en|uk/ do
     root 'campaigns#index'
     devise_for :users, controllers: { registrations: 'users/registrations' }
-    resources :campaigns
-    resources :news
+    resources :news do
+      resources :comments
```

Comment:
You don't have `show` and `new` actions in your controller. Also please read about nested resources, especially nesting limits and shallow nesting http://guides.rubyonrails.org/routing.html#nested-resources